### ISSUE TITLE
Bug: incorrect camelcase 

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16028]
### Environment

Node version: v16.14.2
npm version: v8.12.1
Local ESLint version: Not found
Global ESLint version: v8.17.0 (Currently used)
Operating System: linux 4.18.0-348.7.1.el8_5.x86_64

### What parser are you using?

Default (Espree)

### What did you do?

```js
const FooBar = 'hello';
```

### What did you expect to happen?

Error for camelcase, since correct code would be:

```js
const fooBar = 'hello';
```

### What actually happened?

No error

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @kkmuffme, thanks for the issue!

This works as intended. Per [Rule Details](https://eslint.org/docs/rules/camelcase#rule-details), this rule reports identifiers with underscores in the name. Since `FooBar` does not contain underscores, it's valid for this rule.
